exercises/minesweeper/src/example/kotlin/MinesweeperBoard.kt

```diff
@@ -0,0 +1,67 @@
+private const val MINE_CHAR = '*'
+private const val SPACE_CHAR = ' '
+
+data class MinesweeperBoard(val mineLocations: List<String>) {
+
+    private val numberOfRows by lazy {
+        mineLocations.size
+    }
+
+    private val numberOfColumns by lazy {
+        if (mineLocations.isEmpty()) 0 else mineLocations[0].length
+    }
+
+    fun withNumbers(): List<String> {
+        val result = ArrayList<String>()
+
+        (0 until numberOfRows).mapTo(result) {
+            rowNumber -> getRowWithNumbers(rowNumber)
+        }
+
+        return result
+    }
+
+    private fun getRowWithNumbers(rowNumber: Int): String {
```

Comment:
Could be simplified to: `private fun getRowWithNumbers(rowNumber: Int) = (0..(numberOfColumns - 1)).map { columnNumber -> getCellNumber(rowNumber, columnNumber) }. joinToString("")`